litellm/integrations/langfuse.py

```diff
@@ -81,11 +81,15 @@ def log_event(
             metadata = (
                 litellm_params.get("metadata", {}) or {}
             )  # if litellm_params['metadata'] == None
-            prompt = [kwargs.get("messages")]
             optional_params = kwargs.get("optional_params", {})
 
-            optional_params.pop("functions", None)
-            optional_params.pop("tools", None)
+            prompt = {"messages": kwargs.get("messages")}
+            functions = optional_params.pop("functions", None)
```

Comment:
@webkonstantin can we deepcopy optional params, and then make any pop actions on the deepcopy

Concern - not sure if Logging is deepcopying these params, so this might impact other args

Reply:
done, added deepcopy